### PR TITLE
Backend/mutation/recompute

### PR DESCRIPTION
**Pull request by jeffreyhelgers**
The mutations are now computed each time the user zooms in on the graph. Because the number of nodes on the screen is never that many, the performance of the system is still good.

### CODE REVIEW

**Comment by MatthijsRijlaarsdam** [comment]
When comparing more than 2 genomes, this algorithm still works but i think the node splitting makes it a bit weird detect too many mutations. Let me see if I can find a solution for this.

File: backEnd/src/main/java/ribbonnodes/RibbonController.java [hunk 1]
```diff
@@ -140,72 +141,106 @@ protected void collapseRibbons(ArrayList<RibbonNode> nodes, int zoomLevel) {
...
             }
+
+            level++;
         }
-        return null;
+
+
     }
 
+    protected RibbonNode findPreviousNodeSameGenomes(ArrayList<RibbonNode> nodes, RibbonNode node) {
```
Comment (Icety) [inline backEnd/src/main/java/ribbonnodes/RibbonController.java]: Missing JavaDoc

File: backEnd/src/main/java/ribbonnodes/RibbonNode.java [hunk 2]
```diff
@@ -299,4 +302,38 @@ public boolean isVisible() {
...
+     * Checks if there is a mutation on this node.
+     * @return Boolean type.
+     */
+    public boolean hasMutation() {
+    	return mutations.size() > 0;
+    }
+
+
+
+    public boolean isyFixed() {
```
Comment (Icety) [inline backEnd/src/main/java/ribbonnodes/RibbonNode.java]: Missing JavaDoc

**Comment by MatthijsRijlaarsdam** [comment]
Added lazy loading, lots of performance, added right zooming

**Comment by Icety** [comment]
LGTM

**Pull request merged**